src/__test__/components/plots/styling/frequency/SelectCellSets.test.jsx

```diff
@@ -1,52 +1,91 @@
+
+const mockOnUpdate = jest.fn().mockImplementation(() => { console.error('*** TESTING ***'); });
+
+const defaultProps = {
+  onUpdate: mockOnUpdate,
+  config: initialPlotConfigStates.frequency,
+};
+
+const selectCellSetsFactory = createTestComponentFactory(SelectCellSets, defaultProps);
+
+const cellSetsData = require('__test__/data/cell_sets.json');
+
+let storeState = null;
 
 describe('Select cell sets tests ', () => {
-  const config = initialPlotConfigStates.frequency;
-  const store = mockStore({});
-  const mockOnUpdate = jest.fn();
-
-  const renderSelectCellSets = () => {
-    const optionsMetadata = mockCellSets1.hierarchy.filter((set) => set.key === 'sample');
-    const optionsCellSets = mockCellSets1.hierarchy.filter((set) => set.key !== 'sample');
-    render(
-      <Provider store={store}>
-        <SelectCellSets
-          onUpdate={mockOnUpdate}
-          config={config}
-          optionsMetadata={optionsMetadata}
-          optionsCellSets={optionsCellSets}
-        />
-      </Provider>,
-    );
-  };
-
-  it('renders properly', () => {
-    renderSelectCellSets();
-    expect(screen.getByText('Select the metadata that cells are grouped by', { exact: false })).toBeInTheDocument();
-    expect(screen.getByText('Select the cell sets to be shown:')).toBeInTheDocument();
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    enableFetchMocks();
+    fetchMock.resetMocks();
+    fetchMock.doMock();
+    fetchMock.mockResponse(JSON.stringify(cellSetsData));
+
+    storeState = makeStore();
+
+    storeState.dispatch(loadCellSets(fake.EXPERIMENT_ID));
+  });
+
+  it('Renders properly', async () => {
+    await act(async () => {
+      render(
+        <Provider store={storeState}>
+          {selectCellSetsFactory()}
+        </Provider>,
+      );
+    });
+
+    // 1st select to select sample
+    expect(screen.getByText(/Select the metadata that cells are grouped by/, { exact: false })).toBeInTheDocument();
+    expect(screen.getByRole('combobox', { name: 'metadata' })).toBeInTheDocument();
+
+    // 2nd to select the group by
+    expect(screen.getByText(/Select how the data should be grouped/)).toBeInTheDocument();
+    expect(screen.getByRole('combobox', { name: 'groupBy' })).toBeInTheDocument();
   });
 
   it('switching cellsets updates the plot', async () => {
-    await renderSelectCellSets();
-    const box = screen.getByLabelText('cell sets', { selector: 'input' });
-
-    fireEvent.change(box, { target: { value: 'Scratchpad' } });
-    const option = screen.getByText('Scratchpad');
-    fireEvent.click(option);
-    const option2 = screen.getByText('Louvain');
-    fireEvent.click(option2);
-    expect(mockOnUpdate).toHaveBeenCalledTimes(2);
+    await act(async () => {
+      render(
+        <Provider store={storeState}>
+          {selectCellSetsFactory()}
+        </Provider>,
+      );
+    });
+
+    const dropdown = screen.getByRole('combobox', { name: 'groupBy' });
+
+    await act(async () => {
```

Comment:
Do you need to wrap this in act? Did you get the warning for it?

Reply:
Does it need to be an async function too?

Reply:
Yes, I do get the warning if I don't wrap it in `act` or remove the `async`. I think the `async` is there because there are async functions that are dispatched after firing the event, and to handle it we need to make the callback `async` which is resolved by `await`.